src/components/projects/project-edit-form/project-edit-container/project-edit-container.tsx

```diff
@@ -108,10 +131,39 @@ const ProjectEditContainer: React.FC<EditFormContainerProps> = ({ wbsNum, proj,
     }
   };
 
+  const isValidURL = (url: string | undefined) => {
+    if (url !== undefined) {
+      try {
+        new URL(url);
+        return true;
+      } catch (_) {
+        alert("Invalid URL provided.");
+      }
+    } 
+    else {
+      alert("URL not provided.");
+    }
+    return false;
+  }
+
+  const checkValidity = () => {
+    if (budget < 0) {
+      alert("Budget can't be negative.")
+    }
```

Comment:
this can be done a different and better way by adding a `min={0}` prop to the react bootstrap form control.